web-bluetooth/read-characteristic-value-changed.js

```diff
@@ -1,13 +1,34 @@
+var bluetoothDevice;
 var batteryLevelCharacteristic;
 
 function onReadBatteryLevelButtonClick() {
-  log('Requesting Bluetooth Device...');
-  navigator.bluetooth.requestDevice(
-    {filters: anyDevice(), optionalServices: ['battery_service']})
-  .then(device => {
-    log('Connecting to GATT Server...');
-    return device.gatt.connect();
+  let requestDevice = Promise.resolve();
+  if (!bluetoothDevice) {
+    log('Requesting Bluetooth Device...');
+    requestDevice = navigator.bluetooth.requestDevice(
+      {filters: anyDevice(), optionalServices: ['battery_service']})
+    .then(device => {
+      bluetoothDevice = device;
+    });
+  }
+
+  requestDevice.then(connectDevice)
+  .then(_ => {
+    log('Reading Battery Level...');
+    return batteryLevelCharacteristic.readValue();
   })
+  .catch(error => {
+    log('Argh! ' + error);
+  });
+}
+
+function connectDevice() {
```

Comment:
I called this `connectDevice` in my example, but I wonder if it should actually be `connectDeviceAndCacheCharacteristics`.